agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -128,17 +130,53 @@ def apply
           warn "failed to sync #{service_pod.name} at #{service_pod.deploy_rev}: #{error}"
           warn error
           sync_state_to_master(@container, error)
+          return
         else
           @container_state_changed = false
-          sync_state_to_master(@container)
+        end
 
+        if service_pod.terminated?
           # Only terminate this actor after we have succesfully ensure_terminated the Docker container
           # Otherwise, stick around... the manager will notice we're still there and re-signal to destroy
-          self.terminate if service_pod.terminated?
+          self.terminate
+        elsif service_pod.running? && service_pod.wait_for_port
+          # delay sync_state_to_master until started
+          # XXX: apply() gets called twice for each deploy_rev, and this launches two wait_for_port tasks...
+          async.wait_for_port(service_pod, @container)
```

Comment:
This is a side effect of the  #2766 container `create`/`start` events setting `@container_state_changed = true` and thus triggering two calls to `apply` for each `service_pod.deploy_rev`:

```
Celluloid::Actor 0x1de478ea90
Celluloid::Cell 0x1de478ecac: Kontena::Workers::ServicePodWorker
State: Running (executing tasks)
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/mailbox.rb:63:in `sleep'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/mailbox.rb:63:in `wait'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/mailbox.rb:63:in `block in check'
	/usr/lib/ruby/gems/2.4.0/gems/timers-4.1.1/lib/timers/wait.rb:33:in `while_time_remaining'
	/usr/lib/ruby/gems/2.4.0/gems/timers-4.1.1/lib/timers/wait.rb:11:in `for'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/mailbox.rb:58:in `check'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:155:in `block in run'
	/usr/lib/ruby/gems/2.4.0/gems/timers-4.1.1/lib/timers/group.rb:66:in `wait'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:152:in `run'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:131:in `block in start'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-essentials-0.20.5/lib/celluloid/internals/thread_handle.rb:14:in `block in initialize'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/actor/system.rb:78:in `block in get_thread'
	/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/group/spawner.rb:50:in `block in instantiate'


	Tasks:
	  1) Celluloid::Task::Fibered[call]: sleeping
	      {:dangerous_suspend=>false, :method_name=>:wait_for_port}
		Celluloid::Task::Fibered backtrace unavailable. Please try `Celluloid.task_class = Celluloid::Task::Threaded` if you need backtraces here.


	  2) Celluloid::Task::Fibered[call]: sleeping
	      {:dangerous_suspend=>false, :method_name=>:wait_for_port}
		Celluloid::Task::Fibered backtrace unavailable. Please try `Celluloid.task_class = Celluloid::Task::Threaded` if you need backtraces here.
```

Very annoying to try and fix...